Utils/Dataflow/055_documents2TTL/documents2ttl.py

```diff
@@ -202,11 +230,15 @@ def document_cds(data, doc_iri, cds_attrs):
 
 
 def doi2ttl(doi, doc_iri):
-    """
-    Converting DOI parameter to TTL
-    :param doi: doi from JSON string
+    """ Convert DOI parameter from JSON to TTL.
+
+    :param doi: 'doi' part of the initial JSON
+    :type doi: str, unicode or list
```

Comment:
1) I checked the Sphinx's documentation, and it seems that it actually encourages the use of "or", and not comma.
http://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists
2) There is a possibility that we wouldn't be able to get away from the human-oriented words, see comment below, about returning the same type.

Reply:
I did a few tests, and it seems that the result is pretty much the same regardless of what you put into the type - no type linking is performed by Sphinx, even if it's a single type. Dunno, maybe we have some "do_linking = off" in the preferences.

This means that unless we really want the linking to the types to be performed (I doubt it), then we are free to write whatever we want into `:type:`.

However, while making the research another thought occurred to me: text like `something(A, B, C)` can be confused with a tuple or a function. It gets even worse when actual composite types come into play and you have to look at `something(tuple(A, B))` and `something(tuple(A or B))`. Another reason to use "or" and leave comma for the composite types.

Reply:
@Evildoor
> text like `something(A, B, C)` can be confused with a tuple or a function

Without context -- of course. But if you expect to see `<param_name> (<param_type>)`... well,  everything in the brackets most likely will be interpreted as given parameter type(s), not members of a tuple or arguments of another function. "Most likely" -- because I can't get into other people heads and see how they interpret information ;)

> when actual composite types come into play and you have to look at `something(tuple(A, B))` and `something(tuple(A or B))`. Another reason to use "or" and leave comma for the composite types.

If the idea here is to say "parameter `something` is of type `tuple` with members of type `A` or of type `B`" -- then I'd suggest to write it as `tuple(A), tuple(B)`. If types mixture is possible -- fine, we may say `tuple(A, B)`, but we also can leave it for the user who needs this functionality to find it out. Anyway it does not look as a regular decision, to pass a mixed tuple of numbers and numerix strings. It may happen (if some of them are taken from file, some calculated or specified in the code, etc), but I'd say "try to pass values a they come and hope they will be checked in the function; if something went wrong (or you want to be sure that threre'll be no unexpected effects) -- do some extra preparations in your code before using this function."
No one expects strict typing from Python code, so specified type is "what we expect to get"; yet "it might work with something else, if you are careful enough/if it can automatically be casted to a proper type/etc."

If the idea is to say "parameter `something` is of type `tuple` with first member is [of type] `A` and second member is [of type] `B`" -- then it looks like it's no longer about the parameter type, more like about its value (which is better to be described in the parameter description, not in the `type:` section).

> I checked the Sphinx's documentation, and it seems that it actually encourages the use of "or", and not comma.
http://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists

They say that they can automatically link types if they are separated with "or"; which means that here we would need `type: str or unicode or list`... which does not seem very natural for me. And since (as you say) there seems to be no linking no matter what...that is to say, I see no reason to "leave comma for the composite types".

---

However, for example in Numpy people try to make things more human-readable then type-strict:
https://github.com/numpy/numpy/blob/master/numpy/matlib.py#L18-L25

But I still not convinced, and do not like to see service words like "or" where I'd expect to see type or a list of types. I mean, when I see something like [this](https://docs.scipy.org/doc/numpy/reference/generated/numpy.empty.html#numpy.empty):

> *int or tuple of int*

-- all words equally italic -- I have to do some work to distinguish types from "service" words:

> _**int** or **tuple of int**_

Particularly because of "of" looking much the same as "or" at the first glance ;)

---

Still we need to decide some common approach... or may be just leave things "as is", and let everyone writes as he/she likes. A bit of mess, of course, but I'm just not sure if further discussion will reveal any more "pros and cons"...

So we need to decide a common style

Reply:
I agree with

> may be just leave things "as is", and let everyone writes as he/she likes. A bit of mess, of course, but I'm just not sure if further discussion will reveal any more "pros and cons"...

This looks like a matter of taste, and so far we did not find any decisive arguments for either side. If we encounter more issues with this, the discussion can be resumed.